### DISCUSSION TITLE
Dial timeout Calculation?

### DISCUSSION THREAD

**Discussion by jaswanthikolla** [coredns/coredns#6039] [General]
Following is the excerpt from [forward](https://coredns.io/plugins/forward/#syntax) plugin about dial timeout ( it's 1 second now, but it is used to 100 ms)

> The dial timeout by default is 30s, and can decrease automatically down to 100ms based on early results.

How is Dial timeout automatically decreased to 100 ms, Based on what? Can you explain this calculation.

**Comment by chrisohaver** [comment]
It uses the following algorithm ...

https://github.com/coredns/coredns/blob/47d8ff6d915eb9ee13b71f68c0ce4b8342e6d5aa/plugin/pkg/proxy/connect.go#L20-L32
